File: src/main/java/com/brasil/transparente/api/dto/DisplayableElementDTO.java
package com.brasil.transparente.api.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Builder
@Setter
@Getter
@Schema(description = "Generic element of the Brazilian administrative hierarchy. Depending on the endpoint context, it may represent a Power, Ministry, Agency, or Management Unit.")
public class DisplayableElementDTO {

    @Schema(description = "Unique identifier of the element")
    private Long id;
    @Schema(description = "Official name of the administrative entity")
    private String name;
    @Schema(description = "Total amount effectively paid in the analyzed period, in BRL")
    private Double totalValueSpent;
    @Schema(description = "Percentage of the parent level total represented by this element")
    private Double percentageOfTotal;
    @Schema(description = "Hierarchical level indicator: 1=Federative Unit, 2=Power, 3=Ministry, 4=Agency, 5=Management Unit")
    private Integer level;
}

File: src/main/kotlin/com/brasil/transparente/api/dto/DisplayableElementDTO.kt
package com.brasil.transparente.api.dto

import io.swagger.v3.oas.annotations.media.Schema

@Schema(description = "Elemento genérico da hierarquia administrativa brasileira. Dependendo do contexto do endpoint, pode representar um Poder, Ministério, Órgão ou Unidade Gestora.")
data class DisplayableElementDTO(
    @field:Schema(description = "Identificador único do elemento")
    var id: Long? = null,
    @field:Schema(description = "Nome oficial da entidade administrativa")
    var name: String? = null,
    @field:Schema(description = "Valor total efetivamente pago no período analisado, em BRL")
    var totalValueSpent: Double? = null,
    @field:Schema(description = "Percentual do total do nível pai representado por este elemento")
    var percentageOfTotal: Double? = null,
    @field:Schema(description = "Indicador do nível hierárquico: 1=Unidade Federativa, 2=Poder, 3=Ministério, 4=Órgão, 5=Unidade Gestora")
    var level: Int? = null,
)